### ISSUE TITLE
RESET to a Drawer parent route WITH animation

### ISSUE THREAD

**Issue by baudev** [react-community/react-navigation#2039]
Before all, I have to precise I searched in the different issues as the #1000 , #307 or even StackOverflow.

I have the following code : 
```
/****  STACKNAVIGATOR 2 ****/
const HomeDrawer = StackNavigator({
   HomeDrawerstack: { screen: Home, navigationOptions: { title: 'Accueil',  } },    
  }, { initialRouteName: 'HomeDrawerstack', }    
 );

/****  STACKNAVIGATOR 3 ****/
const AddIdeaDrawer = StackNavigator({
   AddIdeaDrawerRoot: { screen: AddIdea, navigationOptions: { title: 'AddIDea',  } },    
  }, { initialRouteName: 'AddIdeaDrawerRoot', }    
 );
 
/****  DRAWERNAVIGATOR ****/
const DrawerRoutes = {
    HomeRoot: {
      screen: HomeDrawer, 
      navigationOptions: {
          title: 'Home',
      },
    },
    AddIdea: {
      screen: AddIdeaDrawer 
      navigationOptions: { 
        title: 'Add an idea',
      },
    },
};

const DrawerOptions = {
  initialRouteName: 'HomeRoot',
  contentComponent: CustomDrawerContentComponent,  
  drawerWidth: 300,
}; 

export const Drawer = DrawerNavigator(DrawerRoutes, DrawerOptions); 


/****  STACKNAVIGATOR 1 ****/
export const Navigation = StackNavigator({  
   HomeDrawerRoot: { screen: Drawer, navigationOptions: { header: null } },    
   OtherRoot: { screen: Contacts, navigationOptions: { /** **/ } }, 
  }, { initialRouteName: 'HomeDrawerRoot', }    
 ); 
```
 
### Current Behavior

Currently, I'm trying to navigate from HomeDrawerstack to AddIdeaDrawerRoot with the StackNavigator animation and preventing the user to go back.
I have tried different solutions I have found on the different issues but any of them solve this problem.

**First try :** 
`navigate('AddIdea',)}`
The app goes to AddIdea perfectly. However, the user can go back (that's normal I don't use RESET) and there is no animation. The AddIdea screen appears like that without any transition. 

**Second try :** 
```
NavigationActions.reset({
          index: 0,
          actions: [
            NavigationActions.navigate({routeName: 'AddIdea',})
          ]
})
```
It doesn't work. 
The logs prints :  ```There is no route defined for key AddIdea.  Must be one of: 'HomeDrawerstack'``` 

**Third try ** 
I read on an another issue #1127 which explains that the key:null was the solution. So I tried the following action : 
```
NavigationActions.reset({
          index: 0,
          key: null,
          actions: [
            NavigationActions.navigate({
              routeName: 'HomeDrawerRoot',
              action: NavigationActions.navigate({ routeName: 'AddIdea',}),
              })
          ]
 })
```       
The app goes well to AddIdea. However, still  with the possibility of going back to the last page (Home) and without animation.

**Fourth try :** 
```
navigate({
	routeName: 'HomeDrawerRoot',
	action: NavigationActions.navigate({ 
		routeName: 'AddIdea', action: NavigationActions.reset({
  						index: 0,
  						actions: [
    						NavigationActions.navigate({ routeName: AddIdeaDrawerRoot
  						]
						})
            })
})
```
It doesn't work. Printing :  ```There is no route defined for key AddIdeaDrawerRoot.  Must be one of: 'HomeDrawerstack'``` 

### Expected Behavior

The only thing I want is going from HomeDrawerstack (Home screen) to AddIdeaDrawerRoot (AddIdea screen). With an animation like the StackNavigator animation and preventing the user of going back. (Disable Back action). 

After reading a lot of issues, I didn't find any solutions. Is it a forget of the react-navigation ?
### Your Environment

| software         | version
| ---------------- | -------
| react-navigation |  ^1.0.0-beta.11
| react-native     |  0.45.1
| node             | 7.4.0
| npm or yarn      |  0.24.5

**Comment by baudev** [comment]
Anybody knows how to do that ?

**Comment by spencercarli** [comment]
Hi! In an effort to get the hundreds of issues on this repo under control I'm closing issues tagged as questions. Please don't take this personally - it's simply something we need to do to get the issues to a manageable point. If you still have a question I encourage you to re-read the docs, ask on StackOverflow, or ask on the #react-navigation Reactiflux channel. Thank you!

**Comment by GaudamThiyagarajan** [comment]
@spencercarli  looks like this is an issue. Nested actions within reset aren't handled.

```
NavigationActions.reset({
          index: 0,
          key: null,
          actions: [
            NavigationActions.navigate({
              routeName: 'HomeDrawerRoot',
              action: NavigationActions.navigate({ routeName: 'AddIdea'}),
              })
          ]
 }) 
```

 takes to HomeDrawerRoot and ignores the action nested 
`action: NavigationActions.navigate({ routeName: 'AddIdea'})`

**Comment by GaudamThiyagarajan** [comment]
Looks like the issue is fixed in v1.5.8.  #3593
